add PredictKube Scaler docs (#617)
kedacore/keda-docs: content/docs/2.6/scalers/predictkube.md

+++
title = "Predictkube"

> availability = "v1.0+"

Reviewer: ```suggestion
availability = "v2.6+"
```

maintainer = "Dysnix"

> description = "AI-based predictive scaling based on Prometheus metrics."

Reviewer: ```suggestion
description = "AI-based predictive scaling based on Prometheus metrics & PredictKube SaaS."
```

To make our AI model access your data and make a prediction based on it, please use this API key we provide in PredictKube.

Review our [Privacy Policy](https://predictkube.com/privacy-policy) to see how your data circulates in and out PredictKube.

### Trigger Specification

This specification describes the `predictkube` trigger that scales based on a predicting load based on `prometheus` metrics.

```yaml
triggers:
- type: predictkube
  metadata:
    # Required fields:
    predictHorizon: "2h"
    historyTimeWindow: "7d"
    prometheusAddress: http://<prometheus-host>:9090
    metricName: http_requests_total
    query: sum(rate(http_requests_total{deployment="my-deployment"}[2m]))
    queryStep: "2m"
    threshold: '100'
```

**Parameter list:**

- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
- `prometheusAddress` - Address of Prometheus server.

> - `metricName` - Name to identify the Metric in the external.metrics.k8s.io API. API. If using more than one trigger it is required that all `metricName`(s) be unique.

Reviewer 2: As mentioned on the main PR, we are trying to not expose this property for users, unless there is a really good reason.

- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.

> - `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.

Reviewer: What about this?
```suggestion
- `queryWindow` - The maximum time between two slices within the boundaries for QML range query, used in the query.
```

Reviewer 2: I am fine with both 🤷

- `threshold` - Value to start scaling for.

> ### Authentication Parameters

Reviewer 2: Would be great to mention, that Authentication for referenced Prometheus works as well. Noting it here in this section and in the example Secret

```yaml
triggers:
- type: predictkube
  metadata:
    # Required fields:
    predictHorizon: "2h"
    historyTimeWindow: "7d"
    prometheusAddress: http://<prometheus-host>:9090
    query: sum(rate(http_requests_total{deployment="my-deployment"}[2m]))
    queryStep: "2m"
    threshold: '100'
```

**Parameter list:**

- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
- `prometheusAddress` - Address of Prometheus server.
- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
- `threshold` - Value to start scaling for.

### Authentication Parameters

Predictkube Scaler supports one type of authentication - authentication by API key.

> Prometheus supports all authentication methods that are available in Prometheus Scaler.

Reviewer 2: ```suggestion
Prometheus used in Predictkube Scaler supports all authentication methods that are available in Prometheus Scaler.
```

    prometheusAddress: http://<prometheus-host>:9090
    metricName: http_requests_total
    query: sum(rate(http_requests_total{deployment="my-deployment"}[2m]))
    queryStep: "2m"
    threshold: '100'
```

**Parameter list:**

- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
- `prometheusAddress` - Address of Prometheus server.
- `metricName` - Name to identify the Metric in the external.metrics.k8s.io API. API. If using more than one trigger it is required that all `metricName`(s) be unique.
- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
- `threshold` - Value to start scaling for.

### Authentication Parameters

Predictkube Scaler supports one type of authentication - authentication by API key.

**Auth gateway based authentication:**

> - `apiKey` - API key previously issued for this tenant.

Reviewer: API key of what? :)

Author: for PredictKube API. It can be obtained for free at https://predictkube.com/

Reviewer: By now I've figured that out, but this is something that should be as part of the documentation.

Can you add that please?

Author: Sure, we have added comment in https://github.com/dysnix/keda-docs/commit/1ff8f1dc02114b98a7c301f7025e05c33830ecb7

Thank you!

go_file = "predictkube_scaler"
+++

> PredictKube as an open-source project with the SAAS part consisting of an AI model that requires API to connect to the project for the main functions of predicting and scaling.

Reviewer: How about this?
```suggestion
PredictKube is an open-source project that uses a SaaS offering consisting of an AI model consumed by connecting to a REST API for the main functions of predicting and scaling.
```

> To make our AI model access your data and make a prediction based on it, please use this API key we provide below.

Reviewer: This feels more accurate given it is not issued by KEDA, what do you think?
```suggestion
To make our AI model access your data and make a prediction based on it, please use this API key we provide in PredictKube.
```

> Review our [Privacy Policy](https://predictkube.com/privacy-policy) to see how your data circulates inside PredictKube.

Reviewer: As per your policy page:
```suggestion
Review our [Privacy Policy](https://predictkube.com/privacy-policy) to see how your data circulates in and out PredictKube.
```
